client/src/components/screens/PatientSubmission/ClinicalInformation.jsx

```diff
@@ -416,14 +416,20 @@ class ClinicalInformation extends React.Component {
 
   handleHpoNodesChecked(_, info) {
     const { actions, observations } = this.props;
+    const { treeData } = this.state;
 
     const checkedNodes = info.checkedNodes.map(n => ({ code: n.key, display: n.props.title }));
     const hpoResources = observations.hpos;
 
     const toDelete = [];
     const toAdd = [];
     hpoResources.forEach((resource) => {
-      if (checkedNodes.find(r => r.code === resource.valueCodeableConcept.coding[0].code) == null) {
+      const code = resource.valueCodeableConcept.coding[0];
+      const isUnchecked = checkedNodes.find(r => r.code === code) === null;
+      // Resources selected by the autocomplete aren't in the treeData
+      const isHidden = treeData.find(td => td.key === code) === null;
+
+      if (isUnchecked || isHidden) {
```

Comment:
This would introduce a regression where you could never unselect an HPO. 

As it is right now, `isUnchecked` and `isHidden` would always be `false`, since you're comparing a `r.code` (a `string`) with `code` (`object | undefined`), same thing with `td.key === code` (explanation following):  
 
- Right now, you're storing the value of `coding` in `code`, meaning `code` here is either `Coding | undefined`, but what you want is the `coding[0].code` (which is a `string`). So you should extract it as `coding[0].code`, or just add brackets around `code` like `const { code }`.
- While doing that, you also need to add a pre-codition to make sure the coding array is not empty `coding.length > 0`

Reply:
Good catch!